prediction_market_agent/agents/safe_guard_agent/app.py

```diff
@@ -115,27 +115,15 @@ def get_safe_history_cached(
 if transaction_id is None:
     st.stop()
 
-transaction = check_not_none(
-    next(
-        (
-            tx
-            for tx in queued_transactions + historical_transactions
-            if tx.id == transaction_id
-        ),
-        None,
-    )
-)
-
 
 def run_validation() -> None:
     validate_safe_transaction(
-        safe_address_checksum,
-        transaction,
+        check_not_none(transaction_id),
```

Comment:
Refactored validate_safe_transaction back to accept transaction id, because that is required in API endpoint.